test/NuGet.Core.Tests/NuGet.Versioning.Test/NuGetVersionTest.cs

```diff
@@ -363,5 +363,131 @@ public void TryParseReturnsFalseWhenUnableToParseString(string versionString)
             Assert.False(result);
             Assert.Null(version);
         }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("         ")]
+        [InlineData("1beta")]
+        [InlineData("1.2Av^c")]
+        [InlineData("1.2..")]
+        [InlineData("1.2.3.4.5")]
+        [InlineData("1.2.3.Beta")]
+        [InlineData("1.2.3.4This version is full of awesomeness!!")]
+        [InlineData("So.is.this")]
+        [InlineData("1.34.2Alpha")]
+        [InlineData("1.34.2Release Candidate")]
+        [InlineData("1.4.7-")]
+        [InlineData("1.4.7-*")]
+        [InlineData("1.4.7+*")]
+        [InlineData("1.4.7-AA.01^")]
+        [InlineData("1.4.7-AA.0A^")]
+        [InlineData("1.4.7-A^A")]
+        [InlineData("1.4.7+AA.01^")]
+        [InlineData("1.2147483648")]
+        [InlineData("1.1.2147483648")]
+        [InlineData("1.1.1.2147483648")]
+        [InlineData("1.1.1.1.2147483648")]
+        [InlineData("10000000000000000000")]
+        [InlineData("1.10000000000000000000")]
+        [InlineData("1.1.10000000000000000000")]
+        [InlineData("1.1.1.1.10000000000000000000")]
+        [InlineData("1..2")]
+        [InlineData("....")]
+        [InlineData("..1")]
+        [InlineData("-1.1.1.1")]
+        [InlineData("1.-1.1.1")]
+        [InlineData("1.1.-1.1")]
+        [InlineData("1.1.1.-1")]
+        [InlineData("1.")]
+        [InlineData("1.1.")]
+        [InlineData("1.1.1.")]
+        [InlineData("1.1.1.1.")]
+        [InlineData("1.1.1.1.1.")]
+        [InlineData("1     1.1.1.1")]
+        [InlineData("1.1     1.1.1")]
+        [InlineData("1.1.1     1.1")]
+        [InlineData("1.1.1.1     1")]
+        [InlineData(" .1.1.1")]
+        [InlineData("1. .1.1")]
+        [InlineData("1.1. .1")]
+        [InlineData("1.1.1. ")]
+        [InlineData("1 .")]
+        [InlineData("1.1 .")]
+        [InlineData("1.1.1 .")]
+        [InlineData("1.1.1.1 .")]
+        [InlineData("2147483648.2.3.4")]
+        [InlineData("1.2147483648.3.4")]
+        [InlineData("1.2.2147483648.4")]
+        [InlineData("1.2.3.2147483648")]
+        [InlineData("..1.2")]
+        [InlineData("-1.2.3.4")]
+        [InlineData("1.-2.3.4")]
+        [InlineData("1.2.-3.4")]
+        [InlineData("1.2.3.-4")]
+        [InlineData("   1 9")]
+        [InlineData("   19.   1 9")]
+        [InlineData("   19.   19.   1 9")]
+        [InlineData("   19.   19.   19.   1 9")]
+        [InlineData("1 9   ")]
+        [InlineData("19   .1 9   ")]
+        [InlineData("19   .19   .1 9   ")]
+        [InlineData("19   .19   .19   .1 9   ")]
+        [InlineData("   1 9   ")]
+        [InlineData("   19   .   1 9   ")]
+        [InlineData("   19   .   19   .   1 9   ")]
+        [InlineData("   19   .   19   .   19   .   1 9   ")]
+        public void CatchesInvalidVersionStrings(string versionString)
+        {
+            Assert.False(NuGetVersion.TryGetNormalizedVersion(versionString, out var version));
```

Comment:
I think these tests (and the others below), should use `NuGetVersion.Parse` or `NuGetVersion.TryParse` instead.  `TryGetNorializedVersion` is an internal implementation detail that doesn't really matter that much.

Also, a bunch of the test cases are already tested with `NuGetVersion.Parse`, so some of these test cases are duplicates. Adding the new test cases to existing tests will have the same test coverage with fewer tests overall.